build_platform.py

```diff
@@ -29,6 +29,19 @@
     sys.argv.pop(sys.argv.index("--build_timeout") + 1)
     sys.argv.remove("--build_timeout")
 
+# optional wippersnapper argument to generate a dependencies header file
+PRINT_DEPENDENCIES_AS_HEADER = False
+INCLUDE_PRINT_DEPENDENCIES_HEADER = False
+PRINT_DEPENDENCIES_AS_HEADER_FILENAME = None
+if "--include_print_dependencies_header" in sys.argv:
+    # check argument not null and folder path exists
+    PRINT_DEPENDENCIES_AS_HEADER_FILENAME = sys.argv[sys.argv.index("--include_print_dependencies_header") + 1] if len(sys.argv) > sys.argv.index("--include_print_dependencies_header") + 1 else None
```

Comment:
You're re-calculating `sys.argv.index`, it happens here and on L42. Instead, try storing the sys.argv.index into an `index` variable at the top of this conditional